Commhistory implementation.

tswindell opened the pull request.

Commit:
[voicecall] Updated providerId for telepathy accounts to work with commhistory.

Signed-off-by: Tom Swindell <tom.swindell@jollamobile.com>

matthewvogt commented on plugins/commhistory/src/calleventhandler.cpp:
```diff
@@ -0,0 +1,136 @@
+    if(!d->created) WARNING_T("Failed to push event!");
+}
+
+void CallEventHandler::onEventsCommitted(QList<CommHistory::Event> events, bool success)
```
"const QList &events"?
And then Q_UNUSED(events).

matthewvogt commented on plugins/commhistory/src/calleventhandler.cpp:
```diff
@@ -0,0 +1,136 @@
+{
+    TRACE
+    Q_D(CallEventHandler);
+    if(success) d->committed = true;
```
Committed doesn't seem to be used for anything?

matthewvogt commented:
At least one of the commits needs a bug reference.

matthewvogt commented:
LGTM

Commit:
[voicecall] Call based commhistory logging in voicecall. Contributes JB#32370

 As commhistory doesn't recognised CallChannel based call types, which
are used for VoIP applications in telepathy. It was decided to move call
logging out of commhistory, and in to voicecall by the implementation of
this plugin.

Signed-off-by: Tom Swindell <tom.swindell@jollamobile.com>

jpetrell commented:
Sorry totally missed these pull requests getting merged. I work in Phone area, the tasks were marked under VOIP.

This is late, but some comments anyway:
- if call logging is moved to voicecall why is StreamChannelListener still run on CommHistory side?
- outbound calls also get marked as missed calls, which is behavior a break
- missed call counter is broken, after each missed call the eventCount value increases by two, I wonder if it is because of two parties logging calls (StreamChannelListener, CallEventHandler)
- did testing team do sanity testing / execute call tests?

jpetrell commented:
Reverted for now. Created pull request to re-enable https://git.merproject.org/mer-core/voicecall/merge_requests/5 , also fixed the "outbound calls also get marked as missed calls" issue.
